### PR TITLE
[flang][CMake] Add missing dependency to generate Fortran module files

### PR DESCRIPTION
**Pull request by mjklemm**
Fixes bug https://github.com/llvm/llvm-project/issues/90769.  Many thanks to @Meinersbur for providing the initial thought and solution to this.

### CODE REVIEW

**Review by Meinersbur — APPROVED** [review]

File: llvm/runtimes/CMakeLists.txt [hunk 1]
```diff
@@ -432,7 +432,7 @@ if(runtimes)
       # TODO: This is a workaround until flang becomes a first-class project
       # in llvm/CMakeList.txt.  Until then, this line ensures that flang-new is
       # built before "openmp" is built as a runtime project.
-      list(APPEND extra_deps "flang-new")
+      list(APPEND extra_deps "flang-new" "module_files")
```
Comment (kiranchandramohan) [inline llvm/runtimes/CMakeLists.txt:435]: Could you add a comment on why the `module_files` is required here?

Comment (mjklemm) [reply llvm/runtimes/CMakeLists.txt:437]: Done!

**Review by kiranchandramohan — APPROVED** [review]
LG.

**Comment by mjklemm** [comment]
The issue was present in runtime builds of OpenMP, but was more like a race condition with some succesful builds and some failing.  Esp. for stage2 builds it failed more consistently.  I guess regular project build were not affected, as all targets had been build in order.

**Pull request merged**